### PR TITLE
feat(guild): add `fetch_role`

### PR DESCRIPTION
**Pull request by spifory**
## Summary

- Implements discord/discord-api-docs#7074

Tested with
```py
from nextcord import Intents, Interaction, Role
from nextcord.ext.commands import Bot

bot = Bot(
    intents=Intents(guilds=True, message_content=True, messages=True),
    default_guild_ids=[...],
)

@bot.slash_command()
async def role(inter: Interaction, role: Role):
    assert inter.guild is not None
    fetched_role = await inter.guild.fetch_role(role.id)
    return await inter.send(repr(fetched_role))

bot.run(...)

```
image

## This is a **Code Change**

- [x] I have tested my changes.
- [x] I have updated the documentation to reflect the changes.
- [x] I have run `task pyright` and fixed the relevant issues.

### CODE REVIEW

File: nextcord/guild.py [hunk 1]
```diff
@@ -2804,6 +2804,32 @@ async def fetch_roles(self, *, cache: bool = False) -> List[Role]:
...
+        Parameters
+        ----------
+        role_id: int
+            The ID of the role to fetch.
+
+        Raises
+        ------
+        HTTPException
+            Retrieving the roles failed.
+
```
Comment (xXenvy) [inline nextcord/guild.py:2823]: I've tested, it actually throws an exception like this
```suggestion
        Raises
        ------
        HTTPException
            Retrieving the roles failed.
        NotFound
            The role was not found.
```

File: nextcord/guild.py [hunk 2]
```diff
@@ -2804,6 +2804,32 @@ async def fetch_roles(self, *, cache: bool = False) -> List[Role]:
...
+        Returns
+        -------
+        :class:`Role`
+            The role in the guild.
+        """
+        data = await self._state.http.get_role(self.id, role_id)
+
```
Comment (xXenvy) [inline nextcord/guild.py:2830]:
```suggestion
```

**Review by xXenvy — CHANGES_REQUESTED** [review]
otherwise looks good

File: nextcord/guild.py [hunk 3]
```diff
@@ -2804,6 +2804,32 @@ async def fetch_roles(self, *, cache: bool = False) -> List[Role]:
 
         return roles
 
+    async def fetch_role(self, role_id: int, /) -> Role:
+        """|coro|
+
+        Retrieves a :class:`Role` that a guild has.
```
Comment (ooliver1) [inline nextcord/guild.py:2810]: I understand you're following `fetch_roles` wording but to be picky I would prefer it to be similar to fetching single objects by ID, something like:

```suggestion
        Retrieve a :class:`Role` from this guild by its ID.
```
also matches the imperative form docstrings should have (which isn't true for all yet)

Comment (spifory) [reply nextcord/guild.py:2810]: Since this function under the `Guild` class, is specifying "guild by its ID" not a bit redundant?

Comment (ooliver1) [reply nextcord/guild.py:2810]: "A role the guild has" would also be redundant in that case, you can put what you like as there's no real convention yet, but imperative and descriptive is the idea

**Review by xXenvy — APPROVED** [review]

**Review by ooliver1 — APPROVED** [review]

**Pull request merged**